core/src/main/java/org/elasticsearch/index/store/FsDirectoryService.java

```diff
@@ -127,6 +130,7 @@ protected Directory newFSDirectory(Path location, LockFactory lockFactory) throw
         throw new IllegalArgumentException("No directory found for type [" + storeType + "]");
     }
 
+    @Deprecated
```

Comment:
I did it to make it obvious that this code should be removed but I am fine with your suggestion too. I will update.